pkg/generated/openapi/cmd/models-schema/main.go

```diff
@@ -42,6 +43,14 @@ func output() error {
 	defs := openapi.GetOpenAPIDefinitions(refFunc)
 	schemaDefs := make(map[string]spec.Schema, len(defs))
 	for k, v := range defs {
+		// replace top-level schema with v2 if v2 schema exists
+		// this is because this program should output OpenAPI v2 spec.
+		if schema, ok := v.Schema.Extensions[common.ExtensionV2Schema]; ok {
+			if v2Schema, isOpenAPISchema := schema.(spec.Schema); isOpenAPISchema {
+				schemaDefs[friendlyName(k)] = v2Schema
+				continue
+			}
+		}
```

Comment:
Can you explain what this is and how it works?

Reply:
Updated the comments to clarify this part of code.